src/components/Create/Create.jsx

```diff
@@ -81,42 +153,93 @@ function Create(props) {
                 ...prevState,
                 name: e.value
             }))
-            setWarningText(null)
+            setNameText(null)
         }
     } 
 
     // handle create catchment
     const handleCreate = async () => {
-        if (formInputs.name === "" ) return
-        if (formInputs.country === "") return
-        // TODO: to create catchment, will need data from DHIS2
-        // {
-        //     name,
-        //     csv,
-        //     country,
-        //     algorithm,
-        //     fields: {
-        //         lat,
-        //         lng,
-        //         name
-        //     }
-        // }
-        await createCatchmentJob(formInputs)
+        if (formInputs.country === "") {
+            setCountryText(i18n.t("Country required"))
+            return
+        }
+        if (formInputs.name === "") {
+            setNameText(i18n.t("Name required"))
+            return
+        }
+        if (formInputs.level === "") {
+            setLevelText(i18n.t("Level required"))
+            return
+        }
+        
+        setIsLoading(true)
+        const resp = await createCatchmentJob(formInputs).catch( async (err) => {
+            setIsLoading(false)
+            const data = JSON.parse(await err.response.text())
+
+            if (data.csv) {
+                const resp = papaparse.parse(data.csv.trim(), { header: true })
+                return { error: resp }
+            } else {
+                return setAlertError({ text: i18n.t(data.message), critical: true })
+            }
+        })
+        
+        // if there are errors then set the error
+        if (resp?.error) {
+            const errors = resp.error.data.filter((data) => data["cc:ErrorMessage"] !== "") 
+            const clean = resp.error.data.filter((data) => data["cc:ErrorMessage"] === "")
+
+            if (clean.length === 0) {
+                setErrorMessage({ message: i18n.t("There are no valid sites."), proceed: false })
+            } else {
+                setErrorMessage({ message: i18n.t("Click proceed if you want to continue. The sites with errors will be removed if you proceed or click cancel to go back.") , proceed: true })
+            }
+            setErrorData({ data: resp.error.data, fields: resp.error.meta.fields, errors: errors})
+            setHasErrors(true)
+        } else {
+            // close the modal
+            close()
+            // toggle to fetch for jobs
+            toggle()
+            setIsLoading(false)
+            setAlert({ text: i18n.t("Your catchment areas are being created. It should be ready in a few minutes")})
```

Comment:
I think `It` should be changed to `They` since `areas` is plural in the prior sentence.

Reply:
This was taken from our application so we can use the translation from there. It's a little confusing because a catchment job has catchment areas and it is referring to the catchment job.